packages/tab/index.tsx

```diff
@@ -19,35 +19,70 @@
 // LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 // OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 // THE SOFTWARE.
-
-import React, {Component} from 'react';
-import PropTypes from 'prop-types';
-import classnames from 'classnames';
-
+import * as React from 'react';
+import * as classnames from 'classnames';
+// TODO: Come back and use .d.ts file
+// @ts-ignore
 import TabIndicator from '@material/react-tab-indicator';
+// No mdc .d.ts files
+// @ts-ignore
 import {MDCTabFoundation} from '@material/tab/dist/mdc.tab';
+import TabRipple, {TabRipple as TabRippleBase} from './TabRipple';
+
+export interface TabProps extends React.HTMLProps<HTMLButtonElement> {
+  active?: boolean;
+  className?: string;
+  isFadingIndicator?: boolean;
+  indicatorContent?: React.ReactNode;
+  minWidth?: boolean;
+  isMinWidthIndicator?: boolean;
+  stacked?: boolean;
+  previousIndicatorClientRect?: ClientRect;
+}
+
+interface TabState {
+  classList: Set<string>;
+  'aria-selected': boolean;
+  tabIndex?: number;
+  activateIndicator: boolean;
+  previousIndicatorClientRect?: ClientRect;
+}
 
-import TabRipple from './TabRipple';
+// https://github.com/material-components/material-components-web-react/issues/528
+// TODO: switch when ripple is converted to TSX
+function isElementTabRipple(element: any): element is TabRippleBase {
+  return !!element;
+}
 
-export default class Tab extends Component {
-  foundation_ = null;
-  tabElement_ = React.createRef();
-  tabContentElement_ = React.createRef();
-  tabIndicator_ = React.createRef();
-  tabRipple_ = React.createRef();
+export default class Tab extends React.Component<TabProps, TabState> {
+  foundation_?: MDCTabFoundation;
+  tabElement_: React.RefObject<HTMLButtonElement> = React.createRef();
+  tabContentElement_: React.RefObject<HTMLSpanElement> = React.createRef();
+  tabIndicator_: React.RefObject<TabIndicator> = React.createRef();
+  // https://github.com/material-components/material-components-web-react/issues/528
+  // TODO: switch when ripple is converted to TSX
+  tabRipple_: React.RefObject<TabRippleBase> = React.createRef();
+
+  static defaultProps: Partial<TabProps> = {
+    active: false,
+    className: '',
+    isFadingIndicator: false,
+    indicatorContent: null,
+    minWidth: false,
+    isMinWidthIndicator: false,
+    stacked: false,
+  };
 
-  state = {
+  state: TabState = {
     'classList': new Set(),
```

Comment:
`classList` is a `DOMTokenList`, which is not constructable.  It looks like `Set` has some similar methods, but not all (e.g. `contains`, `item`).  What is `classList` used for?  Have you asserted that the overlapping set of functionality is all you need (e.g. nobody is going to try to call `contains` on your `Set`)?

This should all be commented.

Moreover, I don't see you using this state in your component.  That smells really badly to me.  Why are you storing something that isn't tied to rendering in `State`?

Reply:
Used between lines 106 and 125.

Its a pattern we have in most components. Since its on state no one externally should call contains on the Set.

Reply:
https://github.com/material-components/material-components-web-react/pull/548